src/Model/Svg/SvgFile.php

```diff
@@ -470,6 +471,17 @@ protected function analyse(): void
         $this->savedLanguages = array_unique($this->savedLanguages);
     }
 
+    /**
+     * Workaround for O(N^2) performance of DOMDocument::getElementsByTagName() T336917
+     *
+     * @param string $name
+     * @return DOMNodeList
+     */
+    private function getElementsByTagName(string $name): DOMNodeList
```

Comment:
I know it won't, given the simplicity of the calls to this, but technically this can also return `false` if `$name` contains invalid characters etc.

Phan doesn't complain about it though, and we know it's fine, so I guess I'm just nitpicking! :-)

Reply:
If $name contains invalid characters, throwing an exception is appropriate, which is what it will do if there is a return type mismatch. So it seems correct to me.

Reply:
Yep, that makes sense.